lib/matplotlib/widgets.py

```diff
@@ -1237,35 +1245,107 @@ def _get_data(self, event):
         ydata = min(y1, ydata)
         return xdata, ydata
 
+    def _clean_event(self, event):
+        """Clean up an event
+
+        Use prev event if there is no xdata
+        Limit the xdata and ydata to the axes limits
+        Set the prev event
+        """
+        if event.xdata is None:
+            event = self._prev_event
+        else:
+            event = copy.copy(event)
+        event.xdata, event.ydata = self._get_data(event)
+
+        self._prev_event = event
+        return event
+
     def press(self, event):
-        """Button press handler"""
+        """Button press handler and validator"""
         if not self.ignore(event):
-            self.eventpress = copy.copy(event)
-            self.eventpress.xdata, self.eventpress.ydata = (
-                self._get_data(event))
+            event = self._clean_event(event)
+            self.eventpress = event
+            self._prev_event = event
+            key = event.key or ''
+            if 'alt' in key or key == ' ':
```

Comment:
Any reason for allowing two keys here? I'd just pick the space character and go with it.

Tangentially, I am not a fan of these hard-coded special keys. While they are fine as defaults, an application developer may already have these keys mapped to something and would want something different. So, it might be a good idea to have them able to be defined in the constructor.